plugins/custom-codecs/src/main/java/org/opensearch/index/codec/customcodecs/ZstdCodec.java

```diff
@@ -49,4 +59,9 @@ public String toString() {
     public boolean supports(Setting<?> setting) {
         return setting.equals(EngineConfig.INDEX_CODEC_COMPRESSION_LEVEL_SETTING);
     }
+
+    @Override
+    public Set<String> aliases() {
```

Comment:
Minor, for `zstd`, the name of the codec is the same, we may not need `CodecAliases` for it

Reply:
The codec name is  ```Lucene95CustomCodec``` for ```zstd``` as well.

Reply:
Just noticed the other change that it is using ```mode.getCodec()``` as the codec name. @sarthakaggarwal97 - Would this break the segments already written with ``` Lucene95CustomCodec``` as codec name?

Reply:
@backslasht @reta, I'll add back Lucene95CustomCodec as the codec, since by our definition as well, zstd / no-dict are just the modes.

Reply:
I think I am wrong, both of the modes can't be registered Lucene95CustomCodec since in the availableCodecs, only one of ```zstd``` and ```zstd_no_dict``` will reflect with the Lucene95CustomCodec

Reply:
But it won't be backward compatible if I change the codec name to ```zstd``` and ```zstd_no_dict``` as well as mentioned by @backslasht 
Looking for a solution

Reply:
Yeah, I am wondering now what is stored as the codec name for the index since both `zstd` and  `zstd_no_dict` had the same codec name - it is impossible to tell which one is behind without looking into mode, but is mode stored somewhere?

Reply:
Yes, it is stored in the segment infos, here is an example. 
```
                            "_b": {
                                "generation": 11,
                                "num_docs": 13685,
                                "deleted_docs": 0,
                                "size_in_bytes": 207198,
                                "memory_in_bytes": 0,
                                "committed": true,
                                "search": true,
                                "version": "9.8.0",
                                "compound": false,
                                "attributes": {
                                    "Lucene95CustomStoredFieldsFormat.mode": "ZSTD"
                                }
                            }
```
Lucene95CustomCodec is registered with Lucene, but within OpenSearch we know if the codec is ```zstd``` or ```zstd_no_dict``` as a part of index settings

Reply:
So the `Lucene95CustomStoredFieldsFormat` didn't change, this is good, we should be able to properly read the index that was created in 2.9.x, right?

Reply:
Yes, I've added the Lucene95CustomCodec (with a deprecated annotation) to maintain backward compatibility. Going forward, ZSTD and ZSTDNODICT codecs will be used.